lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -424,8 +424,11 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
         b.create<arith::ConstantOp>(loc, b.getFloatAttr(floatDtype, 0));
     return createEqual(b, loc, floatDtype, self, zero);
   }
-  if (isa<AtenAbsOp>(op))
+  if (isa<AtenAbsOp>(op)) {
+    if (payloadArgs[0].getType().isa<IntegerType>())
+      return b.create<math::AbsIOp>(loc, payloadArgs[0]);
     return b.create<math::AbsFOp>(loc, payloadArgs[0]);
+  }
```

Comment:
Are you sure that apart from integer or float it doesn't take any other element type?

Reply:
I am not sure, but I will check. All I know is that before it was only lowering to AbsFOp.

Reply:
MLIR only has math::AbsIOp/AbsFOp here https://mlir.llvm.org/docs/Dialects/MathOps/, so it should only take integer or float element types.